### ISSUE TITLE
Not working

### ISSUE THREAD

**Issue by utbuiquoc** [jeffshee/gnome-ext-hanabi#13]
[Screencast from 09-08-2022 17:19:53.webm](https://user-images.githubusercontent.com/86715840/183625230-6ee6e2d5-e8f4-4323-8123-e86b9df53865.webm)
Thanks for replayed, issue is fixed but video still not showed.

_Originally posted by @utbuiquoc in https://github.com/jeffshee/gnome-ext-hanabi/issues/11#issuecomment-1209196272_

**Comment by jeffshee** [comment]
Hmm, is this the same issue as #2 ?
Please check if your log is similar to that issue. You can get the log by executing `log.sh` that comes with the extension.
What if you use a video with sound? Does the audio play at least?

**Comment by utbuiquoc** [comment]
Here is my log
![image](https://user-images.githubusercontent.com/86715840/183660433-f26a96f9-654b-4099-a779-2196d03af5d8.png)
The video was used here (https://www.desktophut.com/PC-HD-Elaina-Winter-Live-Anime-Wallpaper)

**Comment by jeffshee** [comment]
Thanks for the log. It seems fine to me.
Please also try another video with audio if possible. If you don't hear any audio, it means the video player isn't working in the first place.

**Comment by utbuiquoc** [comment]
In my case, no audio.
Hope this info can help you.

**Comment by jeffshee** [comment]
Hi @utbuiquoc,
and also other users who had reported similar issues @Blaze-Lua @Seikm @CharmedGoose

Sorry to keep you guys waiting~
In the last commit, I added a test script named `test-renderer.sh` for the renderer. 
Ideally, it should spawn a window and play the video, as a regular video player does.
![Screenshot from 2022-08-21 15-00-15](https://user-images.githubusercontent.com/25530920/185777891-b2cf0788-ae71-4aa3-b468-5b18f3ecd1e0.png)

To use the test script, navigate to the `src` directory inside the cloned repo. Execute the script, `./test-renderer.sh`
(Note: A video file named `video.mp4` is required. You don't have to install the extension for the test. See README)

If for some reason the test script doesn't work, it might be the reason why the extension failed to show the live wallpaper.
In that case, please provide the log/error message if any.
Also, I'm not sure whether this issue is related to multimedia codecs or not. I need to narrow down the cause I guess. =)

Looking forward to the test results~ 😎

**Comment by Seikm** [comment]
@jeffshee I ran `./test-renderer-sh`, and it worked normally. I installed the extension normally and for some reason it worked perfectly.

I don't know if you made any other significant changes to the code, but this time it worked out of the box.

I remember that a while ago I tried and it didn't work, I don't know why this time it worked. Maybe I removed some program/library that was in conflict, but I can't remember sorry.

There was a problem on my second monitor, the wallpaper is stretched when I enter activities mode (on the first monitor it works normally).
![image](https://user-images.githubusercontent.com/37481700/185794651-91f91e66-dc6f-4088-9191-a89e9483e2b1.png)

Anyway, thanks for the help and I look forward to new additions to the project.

**Comment by jeffshee** [comment]
@Seikm Great to see that your problem is fixed. 😄

Regarding the problem with the second monitor, is it possible that your monitor is widescreen? (not in the same ratio as the video) If it is just a normal screen (16:9), it is a bug then.

**Comment by Cry4775** [comment]
Hello, I've got the same problem as #2, but in my case i seem to not get it solved. I'll attach the log and a screenshot of the test-renderer.sh 

![Schermata del 2022-08-25 12-36-01](https://user-images.githubusercontent.com/63257144/186643560-cafc59a8-adfc-4ed7-bf96-8d18920882ea.png)
![Schermata del 2022-08-25 12-37-25](https://user-images.githubusercontent.com/63257144/186643586-50e9d504-8fab-4ff6-afa3-bfcca6388081.png)

**Comment by jeffshee** [comment]
@Cry4775 Thanks for reporting.

I'm not too sure, but I'm guessing that this could be a distro-specific issue. 
Let me try to replicate this issue on a PopOS VM.

Meanwhile, could you please try installing the [multimedia codecs](https://support.system76.com/articles/codecs/) and check if the renderer still cannot play your video?
Also, it will be awesome if you could share more details about your PopOS version, Gnome Shell version, X11/Wayland, etc.

Thanks =)

**Comment by jeffshee** [comment]
Alright, for those who have issues with the video doesn't play, I think I had found the cause! 😎
I noticed that the media backend for gtk4 isn't preinstalled in some distros (for example PopOS).

![Screenshot from 2022-08-27 04-50-02](https://user-images.githubusercontent.com/25530920/187012097-649a1802-4d68-4154-9a8e-7a79eed12614.png)

The solution is quite simple. You just need to install the backend:
`sudo apt install libgtk-4-media-gstreamer`

Kindly report if the solution works. Thanks!
